### PR TITLE
Remove update-post-status mixin. Draft: Switch to equivalent HoC

### PR DESCRIPTION
**Pull request by mcsf**
Continuation of #15160, which introduced the `updatePostStatus` HoC and applied it to `Page`. This PR brings analogous changes to `Drafts`, the last remaining consumer of the mixin. Hence, this eliminates the mixin as well.

@mtias, per https://github.com/Automattic/wp-calypso/pull/11593#issuecomment-283678760 it seems that all actions in Drafts are effectively hidden from users. I was tempted to remove all the effectful code and the hidden presentational one, but I'll leave that decision up to you. This component could be greatly, greatly shrunk if all that went away.

Also, note that, although I hooked the new HoC where the mixin was used, some of the existing logic was manually calling `lib/posts/actions`. Whether you decide to remove all the effectful logic or improve it, I suggest starting by removing that `lib/posts/actions` dependency and sticking to the HoC as interface.

### CODE REVIEW

File: client/lib/mixins/update-post-status/README.md [hunk 1]
```diff
@@ -1,36 +0,0 @@
-update-post-status
```
Comment (spen) [inline client/lib/mixins/update-post-status/README.md]: Would it be worth making some amendments to this doc and moving it along side `components/update-post-status/index.js` at this point?

File: client/my-sites/draft/index.jsx [hunk 2]
```diff
@@ -1,78 +1,75 @@
...
 import Notice from 'components/notice';
-import { getSite } from 'state/sites/selectors';
+import NoticeAction from 'components/notice/notice-action';
+import PopoverMenu from 'components/popover/menu';
+import PopoverMenuItem from 'components/popover/menu-item';
+import PostRelativeTimeStatus from 'my-sites/post-relative-time-status';
+import SiteIcon from 'blocks/site-icon';
+import actions from 'lib/posts/actions';
+import photon from 'photon';
+import touchDetech from 'lib/touch-detect';
```
Comment (spen) [inline client/my-sites/draft/index.jsx]: Typo on 'Detech'?

File: client/lib/mixins/update-post-status/README.md (hunk 1 above)
Comment (mcsf) [inline client/lib/mixins/update-post-status/README.md]: Good idea.

File: client/my-sites/draft/index.jsx (hunk 2 above)
Comment (mcsf) [inline client/my-sites/draft/index.jsx]: Ouch. :D

File: client/my-sites/draft/index.jsx [hunk 3]
```diff
@@ -118,35 +113,35 @@ const Draft = React.createClass( {
 		if ( utils.userCan( 'delete_post', this.props.post ) ) {
 			actions.restore( this.props.post, updateStatus );
 		}
-	},
+	}
 
-	previewPost: function() {
+	previewPost = () => {
 		window.open( utils.getPreviewURL( this.props.post ) );
```
Comment (gwwar) [inline client/my-sites/draft/index.jsx]: This existed before, but mind if we add a quick windows check here?

**Comment by gwwar** [comment]
This tests pretty well for me 👍 Should be fine to 🚢  after implementing @spen's suggestions.

**Comment by mcsf** [comment]
Thanks for your feedback, both of you. I'm back from 🌴 and have made the changes. Waiting for CI before 🚢 !

**Pull request merged**